h2/src/test/org/h2/test/unit/TestIgnoreCatalogs.java

```diff
@@ -0,0 +1,158 @@
+package org.h2.test.unit;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.h2.test.TestBase;
+import org.h2.test.TestDb;
+
+/**
+ * @author aschoerk
+ */
+public class TestIgnoreCatalogs extends TestDb {
+    /**
+     * Run just this test.
+     *
+     * @param a ignored
+     */
+    public static void main(String... a) throws Exception {
+        TestBase.createCaller().init().test();
+    }
+
+    @Override
+    public void test() throws Exception {
+        canUseDefaultSchema();
+        canYetIdentifyWrongCatalogName();
+        canUseSettingInUrl();
+        canUseSetterSyntax();
+        canCatalogNameEqualSchemaName();
+        canUseCatalogAtIndexName();
+        canCommentOn();
+        canAllCombined();
+    }
+
+    public void canCommentOn() throws Exception {
+        try (Connection conn = getConnection("jdbc:h2:mem:testdb;IGNORE_CATALOGS=TRUE;init=drop all objects\\;create schema dbo\\;set schema dbo\\;")) {
```

Comment:
You shouldn't use the complete JDBC URLs here and below. We need to test this functionality in all modes. They use different URLs.

Also you need to delete database after execution of `TestIgnoreCatalogs`, because in the most modes local files will be created.

Reply:
At the beginning I tried to test the dbsetting and saw no reasonable way.
I tried to avoid the leftover dbs by using mem. But I will look deeper into the other tests, how to handle that.

Reply:
You can use `getConnection("ignoreCatalogs;IGNORE_CATALOGS=TRUE")`.

Reply:
And `deleteDb("ignoreCatalogs");` before and after your test cases.

Reply:
That's not fixed yet. I need to look deeper into your request.

Reply:
fixed